OpenRA.Mods.Common/Traits/AmmoPool.cs

```diff
@@ -70,21 +79,23 @@ public AmmoPool(Actor self, AmmoPoolInfo info)
 		public bool FullAmmo() { return currentAmmo == Info.Ammo; }
 		public bool HasAmmo() { return currentAmmo > 0; }
 
-		public bool GiveAmmo()
+		public bool GiveAmmo(Actor self, int count)
 		{
 			if (currentAmmo >= Info.Ammo)
 				return false;
 
-			++currentAmmo;
+			currentAmmo += count;
+			UpdateEmptyCondition(self);
 			return true;
 		}
 
-		public bool TakeAmmo()
+		public bool TakeAmmo(Actor self, int count)
 		{
 			if (currentAmmo <= 0)
```

Comment:
It might be a good idea to add `|| count < 0` here so that it can't be exploited to add ammo.